keras/metrics.py

```diff
@@ -34,7 +34,7 @@ def categorical_accuracy(y_true, y_pred):
 
 
 def sparse_categorical_accuracy(y_true, y_pred):
-    return K.cast(K.equal(K.max(y_true, axis=-1),
+    return K.cast(K.equal(K.flatten(y_true),
```

Comment:
When is this required?

Reply:
This is just to make sure y_true passed in is in the correct shape. (e.g. y_true = [0, 1, 1] in my example). However I found a unit test [here](https://github.com/keras-team/keras/blob/master/tests/keras/legacy/interface_test.py#L846), it's passing y_true in the shape of `(number of samples, 1)`. so it's like `y_true=[[0], [1], [1]]`. 

I didn't find any documentation specifying the required label shape. So I assumed both are OK, hence added flatten. If this is not the case, please advice.

Reply:
Got it, thank you. Can we add a small comment explaining that?